migration/migration_test.go

```diff
@@ -0,0 +1,48 @@
+package migration
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/almighty/almighty-core/configuration"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/jinzhu/gorm"
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConcurrentMigrations(t *testing.T) {
+	resource.Require(t, resource.Database)
+
+	var err error
+	if err = configuration.Setup(""); err != nil {
+		panic(fmt.Errorf("Failed to setup the configuration: %s", err.Error()))
+	}
+
+	var wg sync.WaitGroup
+
+	for i := 0; i < 20; i++ {
+		wg.Add(1)
+		db, err := gorm.Open("postgres",
+			fmt.Sprintf("host=%s port=%d user=%s password=%s DB.name=%s sslmode=%s",
+				configuration.GetPostgresHost(),
+				configuration.GetPostgresPort(),
+				configuration.GetPostgresUser(),
+				configuration.GetPostgresPassword(),
+				configuration.GetPostgresDatabase(),
+				configuration.GetPostgresSSLMode(),
+			))
+		if err != nil {
+			t.Fatal("Cannot connect to DB", err)
+		}
+
+		go func(db *gorm.DB) {
+			defer wg.Done()
+			err = Migrate(db)
```

Comment:
Good catch, the migration is executed [before the integration tests start](https://github.com/almighty/almighty-core/pull/278/commits/95ab16af022b4349df33967e427c75b02e440572#diff-552b9cf9c536f6f705ab73d1ab26b005R122) because the integration tests need the common type to be populated. I have to find another way to handle this.